.gitignore

```diff
@@ -15,8 +15,12 @@
 # User-specific stuff:
 .idea/workspace.xml
 .idea/tasks.xml
+.idea/findbugs-idea.xml
+.idea/codeStyleSettings.xml
 .idea/dictionaries
+.idea/shelf
 *.iws
+*.ipr
```

Comment:
We do not use the old file-based format for project (ipr & iws), so both of them could be ignored.
https://www.jetbrains.com/idea/help/project.html

Directory-based format is used instead of it (iml & .idea).